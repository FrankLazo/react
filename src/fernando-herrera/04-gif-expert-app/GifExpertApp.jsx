import { useState } from 'react';
import CategoryAdd from './components/CategoryAdd';
import GifGrid from './components/GifGrid';
import './GifExpertApp.sass';
import 'animate.css';

const GifExpertApp = ({ defaultCategories=[] }) =>
{
	const [categories, setCategories] = useState(defaultCategories);
	// const [categories, setCategories] = useState(['Megadeth']);

	return (
		<div className="gifexpertapp content has-background-dark is-flex is-flex-direction-column is-align-items-start p-6 animate__animated animate__fadeIn">
			<h1 className="has-text-primary">Gif expert app</h1>

			<CategoryAdd setCategories={ setCategories } />

			<hr className="full-rule has-background-grey-dark"/>

		{
			categories.map(category => {
				return (
					<GifGrid
						key={ category }
						category={ category }
						/>
				)
			})
		}
		</div>
	);
};

export default GifExpertApp;
