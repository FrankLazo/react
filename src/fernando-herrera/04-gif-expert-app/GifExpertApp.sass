@import "~bulma/sass/utilities/_all"

.gifexpertapp
	min-height: 100vh

.full-rule
	width: 100%

.card-grid
	display: flex
	flex-wrap: wrap

	img
		max-height: 12rem
		width: auto
