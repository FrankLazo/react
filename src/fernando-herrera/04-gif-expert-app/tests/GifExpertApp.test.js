import { shallow } from 'enzyme';
import GifExpertApp from "../GifExpertApp";

describe('Tests on <GifExpertApp />', () => {
	test('should show the right component', () => {
		const wrapper = shallow(<GifExpertApp />);
		expect(wrapper).toMatchSnapshot();
	});

	test('should show a categories list', () => {
		const categories = ['Megadeth', 'Metallica'];
		const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GifGrid').length).toBe(categories.length);
	});
});
