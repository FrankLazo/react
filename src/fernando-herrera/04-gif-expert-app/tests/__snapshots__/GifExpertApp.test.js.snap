// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifExpertApp /> should show a categories list 1`] = `
<div
  className="gifexpertapp content has-background-dark is-flex is-flex-direction-column is-align-items-start p-6 animate__animated animate__fadeIn"
>
  <h1
    className="has-text-primary"
  >
    Gif expert app
  </h1>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr
    className="full-rule has-background-grey-dark"
  />
  <GifGrid
    category="Megadeth"
    key="Megadeth"
  />
  <GifGrid
    category="Metallica"
    key="Metallica"
  />
</div>
`;

exports[`Tests on <GifExpertApp /> should show the right component 1`] = `
<div
  className="gifexpertapp content has-background-dark is-flex is-flex-direction-column is-align-items-start p-6 animate__animated animate__fadeIn"
>
  <h1
    className="has-text-primary"
  >
    Gif expert app
  </h1>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr
    className="full-rule has-background-grey-dark"
  />
</div>
`;
